packages/dds/tree/src/test/domains/json/benchmarks.ts

```diff
@@ -3,25 +3,43 @@
  * Licensed under the MIT License.
  */
 
-import { fail, strict as assert } from "assert";
-import { ITreeCursor, reduceField, TreeNavigationResult } from "../../../forest";
-import { EmptyKey, FieldKey } from "../../../tree";
+import { forEachNode, ITreeCursorNew } from "../../../forest";
 
-export function sum(cursor: ITreeCursor): number {
+export function sum(cursor: ITreeCursorNew): number {
     let total = 0;
     const value = cursor.value;
     if (typeof value === "number") {
         total += value;
     }
-    for (const key of cursor.keys) {
-        total += reduceField(cursor, key, total, sum);
+    let moreFields = cursor.firstField();
+    while (moreFields) {
+        let inField = cursor.firstNode();
+        while (inField) {
+            total += sum(cursor);
+            inField = cursor.nextField();
+        }
+        moreFields = cursor.nextField();
+    }
```

Comment:
interestingly, I think we can write this as:
```suggestion
    for (let moreFields = cursor.firstField(); moreFields; moreFields = cursor.nextField()) {
        for (let inField = cursor.firstNode(); inField;  inField = cursor.nextField()) {
            total += sum(cursor);
        }
    }
```
 If the performance, it the same (which I suspect it is), this might be better, since it may be easier to follow.

Do you find this more readable?

Reply:
If we like this better, there are a bunch of other places to update as well.

Reply:
I personally find it a lot easier to read. I'll make the updates and make sure it doesn't impact performance.

Reply:
It actually sped things up a bit :)